### PR TITLE
config: remove CURL_SIZEOF_CURL_OFF_T use only SIZEOF_CURL_OFF_T

### PR DESCRIPTION
**Pull request by bagder**
Make the code consistently use a single name for the size of the `curl_off_t` type.

### CODE REVIEW

**Comment by bagder** [comment]
The [travis cmake build](https://travis-ci.org/github/curl/curl/jobs/761931100) now fails to set the largefile bit, but it works fine for me locally. :confused:

**Comment by mback2k** [comment]
The same is true for [these AppVeyor builds](https://ci.appveyor.com/project/curlorg/curl/builds/38112489) due to:
```
test 1014...[Compare curl --version with curl-config --features]
  Mismatch in features lists:
  curl:        alt-svc asynchdns ipv6 ntlm
  curl-config: alt-svc asynchdns ipv6 largefile ntlm
   postcheck FAILED
```

This is the same issue as I experienced with my attempt in https://github.com/curl/curl/pull/6277.

**Comment by mback2k** [comment]
I would actually consider this a fix instead of a tidy-up, because previously Largefile support was not detected at all with CMake, because `CURL_SIZEOF_CURL_OFF_T` is not available in that, just `SIZEOF_CURL_OFF_T`.

**Comment by bagder** [comment]
@snikulov Can you bring any lights to this issue?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1360,8 +1360,8 @@ _add_if("brotli"        HAVE_BROTLI)
 _add_if("zstd"          HAVE_ZSTD)
 _add_if("AsynchDNS"     USE_ARES OR USE_THREADS_POSIX OR USE_THREADS_WIN32)
 _add_if("IDN"           HAVE_LIBIDN2)
-_add_if("Largefile"     (CURL_SIZEOF_CURL_OFF_T GREATER 4) AND
-                        ((SIZEOF_OFF_T GREATER 4) OR USE_WIN32_LARGE_FILES))
+_add_if("Largefile"     (SIZEOF_CURL_OFF_T GREATER "4") AND
+                        ((SIZEOF_OFF_T GREATER "4") OR USE_WIN32_LARGE_FILES))
```
Comment (mback2k) [inline CMakeLists.txt:1364]: Hm, I don't think this is correct. The check worked without quotes in CMake, but `version.c` is somehow not detecting Largefile support. It seems like the detected `SIZEOF_CURL_OFF_T` may not be passed to `curl_config.h` correctly.

Comment (bagder) [reply CMakeLists.txt:1364]: oh right...

**Comment by jay** [comment]
Alternative take at #6704

**Comment by bagder** [comment]
With the new debug output in the cmake travis CI build showing all the SIZEOF sizes after cmake has run shows they're present in `lib/curl_config.h` all right...

**Comment by mback2k** [comment]
What about `USE_WIN32_LARGE_FILES`? Is that present? Can you please link the output in the travis jobs? And I looked at the AppVeyor jobs and unfortunately they either failed, because `#warn` is not understood or CMake build is hiding the output of that.

Looking at the code in `version.c` it should be working... :disappointed:

**Comment by bagder** [comment]
The travis CI cmake [output from the grep](https://travis-ci.org/github/curl/curl/jobs/762034225#L4814-L4822).

**Comment by bagder** [comment]
Even when I reversed the conditional check in `lib/version.c` test 1014 fails the same way on both travis and appveyor!

**Comment by bagder** [comment]
The irony here is of course that I doubt we *ever* build on systems these days that don't support "large files"...

**Comment by bagder** [comment]
I figured it out and it was sillier than you'd expect:

`Largefile` is not generated as a feature for `curl-config` by configure, so therefore, the test 1014 scripts *removes* the word from the curl -V output before it compares it with the `curl-config --features` output...

**Comment by mback2k** [comment]
Wouldn't it be better to fix it the other way around then?

**Comment by bagder** [comment]
Yes, they seem unrelated

#6705 works on the torture test fails
The arm64 tests are flaky since before

**Comment by bagder** [comment]
@jay filed #6704 in the mean time which partly goes in a contrary direction than this so we need to decide.

I like this PR because it both simplifies and unifies the sizeof defines for `curl_off_t`.

**Pull request closed without merging**